### PR TITLE
Add Item Category Support

### PR DESCRIPTION
**Pull request by chadrschroeder**
This pull request:

1. Adds the ability to set categories onto items when creating or updating them.
2. Adds the ability to retrieve the full calendar list so that the category color can be used.

The `CreateItem` and `UpdateItem` requests use a `Categories` element that contains `String` elements that hold category title:

```
<Categories>
   <String/>
</Categories>
```

The new `Exchanger::CategoryString` object will build this `Categories` structure.

The `GetItem` request will continue to return an array of category title `String` values as it has done previously.

If category colors are needed the `categories_with_color` method can be called to perform a `GetUserConfiguration` request to get the category list.

### CODE REVIEW

File: lib/exchanger/elements/category_color.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      22 => [37, 64, 105],    # Dark blue
+      23 => [86, 38, 133],    # Dark purple
+      24 => [128, 39, 93],    # Dark maroon
+    }.freeze
+
+    attr_reader :value, :red, :green, :blue
+
+    def initialize(value)
+      @value = COLORS.keys.include?(value) ? value : -1
+      rgb = COLORS[@value]
```
Comment (ebeigarts) [inline lib/exchanger/elements/category_color.rb]: `rgb = COLORS[value] || COLORS[-1]` would be shorter

File: lib/exchanger/elements/category_color.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      @red = rgb[0]
+      @green = rgb[1]
+      @blue = rgb[2]
+    end
+
+    def css_color
+      "rgb(#{red}, #{green}, #{blue});"
```
Comment (ebeigarts) [inline lib/exchanger/elements/category_color.rb]: I think the color should not include ';'.

**Comment by ebeigarts** [comment]
Everything else looks good to me.

**Comment by chadrschroeder** [comment]
Suggested changes made.

**Comment by ebeigarts** [comment]
Released in 0.2.0

**Pull request merged**